mlrun/artifacts/base.py

```diff
@@ -1275,19 +1278,52 @@ def get_artifact_meta(artifact):
     return artifact_spec, extra_dataitems
 
 
-def generate_target_path(item: Artifact, artifact_path, producer):
-    # path convention: artifact_path[/{run_name}]/{iter}/{key}.{suffix}
+def generate_target_path(
+    item: Union[Artifact, LegacyArtifact], artifact_path: str, producer
+):
+    # path convention: artifact_path[/{run_name}]/{iter}/src_path/{key}.{suffix}
+    if isinstance(item, Artifact):
+        metadata_iter = item.metadata.iter
+        metadata_key = item.metadata.key
+        spec_src_path = item.spec.src_path
+        spec_format = item.spec.format
+    else:  # LegacyArtifact
+        metadata_iter = item.iter
+        metadata_key = item.key
+        spec_src_path = item.src_path
+        spec_format = item.format
```

Comment:
I think this is redundant

Reply:
It is no doubt that it's redundant as long as when someone will edit out the properties of `Artifact` he won't forget to change this as well. It is a reminder, I can delete it if you rather.